kinto_http/tests/test_session.py

```diff
@@ -237,3 +237,11 @@ def test_forced_retry_after_overrides_value_of_header(self):
             session = Session('https://example.org', retry=1, retry_after=10)
             session.request('GET', '/v1/foobar')
             sleep_mocked.assert_called_with(10)
+
+    def test_raises_exception_if_backoff_time_not_spent(self):
+        response = mock.MagicMock()
+        response.status_code = 200
+        self.requests_mock.request.return_value = response
+        session = Session('https://example.org')
+
+        self.assertRaises(BackoffException, session.request, 'get', '/test')
```

Comment:
To be able to do assertion on the exception. In this code we are building an error message: https://github.com/Kinto/kinto-http.py/pull/113/files#diff-052f978a85bf4b9be5b67a6f14ab44d2R23 and it is not trivial to understand what would be the final message. Having it in the test makes it explicit and part of the specification.